docs/sdk/v3/guides/02-quoting.md

```diff
@@ -1,6 +1,113 @@
+
+
+```js reference title="Getting Pool metadata from the Pool smart contact"
+https://github.com/Uniswap/examples/blob/1ef393c2b8f8206a3dc5a42562382c267bcc361b/v3-sdk/quoting/src/example/Example.tsx#L22
+```
+
+The return values of these methods will become inputs to the quote fetching function.
+
+### Setting up a reference to the Quoter contract
+
+Like we did for the Pool contract, we need to construct an instance of an **ethers** `Contract` for our Quoter contract in order to interact with it:
+```js reference title="Setting up a reference to the Quoter contract"
+https://github.com/Uniswap/examples/blob/1ef393c2b8f8206a3dc5a42562382c267bcc361b/v3-sdk/quoting/src/example/Example.tsx#L32
+```
+
+We can get access to the contract's ABI through the [@uniswap/v3-periphery](https://www.npmjs.com/package/@uniswap/v3-periphery) package:
+
+```js reference title="Uniswap V3 Quoter smart contract ABI"
+https://github.com/Uniswap/examples/blob/22ae27bafdbff895ee2168584154626ef4af4d30/v3-sdk/quoting/src/example/Example.tsx#L6
+```
+
+
+### Getting Quotes from the Quoter contract
+
+We can now use our Quoter contact to obtain the quote.
+
+In an ideal world, the quoter functions would be `view` functions, which would make them very easy to query on-chain with minimal gas costs. 
+Instead, the Uniswap V3 Quoter contracts rely on state-changing calls designed to be reverted to return the desired data. 
+This means calling the quoter will be very expensive and should not be called on-chain.
+
+To get around this difficulty, we can use the `callStatic` method provided by **ethers.js** `Contract` instances.
+The same is a useful method that submits a state-changing transaction to an Ethereum node, but asks the node to simulate the state change, rather than to execute it. 
+Our script can then return the result of the simulated state change:
+
+```js reference title="Getting Quotes from the Quoter contract"
+https://github.com/Uniswap/examples/blob/1ef393c2b8f8206a3dc5a42562382c267bcc361b/v3-sdk/quoting/src/example/Example.tsx#L35-L41
+```
+
+It should be noted that `quoteExactInputSingle` is only 1 of 4 different methods that the quoter offers:
+1. `quoteExactInputSingle` - given the amount you want to swap, produces a quote for the amount out for a swap of a single pool
+2. `quoteExactInput` - given the amount you want to swap, produces a quote for the amount out for a swap over multiple pools
+3. `quoteExactOutputSingle` - given the amount you want to get out, produces a quote for the amount in for a swap over a single pool
+4. `quoteExactOutput`  - given the amount you want to get out, produces a quote for the amount in for a swap over multiple pools
+
+## Output
 
-TODO
\ No newline at end of file
+After clicking on the **Quote** button, the output should be available on the running web application.
```

Comment:
This feels like it should also be covered in that setup section instead of the end of the guide. Maybe we start with a `here's the code, follow the README and then this example will do X, Y, and Z if you press button A and B?

That would feel natural to come after the summary of what the guide will cover!

Reply:
Added it to the top!